x-pack/plugins/secops/server/lib/overview/elasticsearch_adapter.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { getOr } from 'lodash/fp';
+
+import { SiemOverviewHostData, SiemOverviewNetworkData } from '../../graphql/types';
+import { FrameworkAdapter, FrameworkRequest, RequestBasicOptions } from '../framework';
+import { TermAggregation } from '../types';
+
+import { buildOverviewHostQuery, buildOverviewNetworkQuery } from './query.dsl';
+import { SiemOverviewAdapter, SiemOverviewHostHit, SiemOverviewNetworkHit } from './types';
+
+export class ElasticsearchSiemOverviewAdapter implements SiemOverviewAdapter {
+  constructor(private readonly framework: FrameworkAdapter) {}
+
+  public async getSiemOverviewNetwork(
+    request: FrameworkRequest,
+    options: RequestBasicOptions
+  ): Promise<SiemOverviewNetworkData> {
+    const response = await this.framework.callWithRequest<SiemOverviewNetworkHit, TermAggregation>(
+      request,
+      'search',
+      buildOverviewNetworkQuery(options)
+    );
+
+    return {
+      packetbeatFlow: getOr(0, 'aggregations.unique_flow_count.doc_count', response),
+      packetbeatDNS: getOr(0, 'aggregations.unique_dns_count.doc_count', response),
+      filebeatSuricata: getOr(0, 'aggregations.unique_suricata_count.doc_count', response),
+      filebeatZeek: getOr(0, 'aggregations.unique_zeek_count.doc_count', response),
+      auditbeatSocket: getOr(0, 'aggregations.unique_socket_count.doc_count', response),
```

Comment:
To the comment above about whether or not to mark these as required -- since you're setting a default for each of these fields I would go ahead and mark them as required then.